src/App.tsx

```diff
@@ -1,26 +1,296 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { deleteTodo, getTodos, patchTodo, postTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import { Footer } from './Footer';
+import { Header } from './Header';
+import { Status } from './enums/status';
+import { Title } from './Title';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [isLoading, setIsLoading] = useState<number | null>(null);
+  const [isEdited, setIsEdited] = useState<number | null>(null);
+  // const [todos, setTodos] = useState<Todo[] | []>([]);
+  const [editedTitle, setEditedTitle] = useState('');
+  const [newTitle, setNewTitle] = useState('');
+  const [errMessage, setErrMessage] = useState('');
+  const [stat, setStat] = useState(Status.all);
+  const [visibleErr, setVisibleErr] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo[] | null>(null);
+  const [todos, setTodos] = useState<Todo[] | []>([]);
+
+  const filteredTodos = (tod: Todo[], type: Status) => {
+    switch (type) {
+      case Status.active:
+        return tod.filter(todo => !todo.completed);
+      case Status.completed:
+        return tod.filter(todo => todo.completed);
+
+      default:
+        return tod;
+    }
+  };
+
+  const visibleTodos = tempTodo
+    ? filteredTodos(tempTodo, stat)
+    : filteredTodos(todos, stat);
+
+  const editSelectedInput = useRef<HTMLInputElement>(null);
+  const isAnyCompleted = todos.some(todo => todo.completed);
+
+  const resetErr = () =>
+    setTimeout(() => {
+      setVisibleErr(false);
+      setErrMessage('');
+    }, 3000);
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      getTodos()
+        .then(setTodos)
+        .catch(() => {
+          setVisibleErr(true);
+          setErrMessage('Unable to load todos');
+          resetErr();
+        });
+    };
+
+    loadTodos();
+  }, []);
+
+  const updateTodo = async (updatedTodo: Todo, option: keyof Todo) => {
+    try {
+      setIsLoading(updatedTodo.id);
+      const todoToUpdate = todos.find(tod => tod.id === updatedTodo.id);
+
+      let newTodo: Todo = {
+        id: 0,
+        userId: 0,
+        title: '',
+        completed: false,
+      };
+
+      if (option === 'completed' && todoToUpdate) {
+        newTodo = { ...todoToUpdate, completed: !updatedTodo.completed };
+      }
+
+      if (option === 'title' && todoToUpdate) {
+        newTodo = { ...todoToUpdate, title: editedTitle };
+      }
+
+      await patchTodo(newTodo);
+
+      await getTodos().then(setTodos);
```

Comment:
Same here, you are sending unnecessary API calls to your backend. 
Backend dev will not be happy :)